### PR TITLE
Use GitHub official CLI to manage releases

### LINKED ISSUE #8409: Use gh command line utility instead of github-release to manage releases
**Description**

Our release pipeline currently downloads a utility called `github-release` from some repo in GitHub. We should instead use the official GitHub CLI which has more features to manage the releases. It's well maintained, actively developed, and more importantly, it's the official way to manage releases via CLI.

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

This PR switches from using a third party github-release tool to using the official GitHub CLI (`gh`) tool to create and manage the release.

In order to always have the latest stable `gh` tool, the GitHub debian repository is added at the beginning in the prepare stage, and gh is installed directly from there.

The github-release.sh script is completely overhauled to use `gh`, with what I hope is improved error handling and helpful logging when things go wrong. The script also now avoids dirtying the state of the repo, in case anything else would need to run later and commit/push things.

Additionally moves creating the Go release tags to the post-github release stage.

## Related issues

closes #8409 
blocked by #8410

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

**Comment by npepinpe** [comment]
Blocked by #8410. One small thing: I haven't fully tested this. By that I mean, I tested the failure cases, and I tested the happy path by having it echo the command it would output (i.e. prepend `echo` in front of the `gh` command at the end), but I didn't dare yet create an actual release. I would propose instead doing that in a fork or something? Cause it will at least create the tags if they don't exist =/

File: .ci/scripts/release/post-github.sh [hunk 1]
```diff
@@ -1,6 +1,11 @@
-#!/bin/bash
+#!/bin/bash -xue
 
-cd "clients/go/internal/embedded"
+# Publish Go tag for the release
+echo git tag "clients/go/v${RELEASE_VERSION}"
```
Comment (menski) [inline .ci/scripts/release/post-github.sh:4]:
```suggestion
git tag "clients/go/v${RELEASE_VERSION}"
```

I assume you actually want to execute the git command?

File: .ci/scripts/release/post-github.sh [hunk 2]
```diff
@@ -1,6 +1,11 @@
...
+echo git push origin "clients/go/v${RELEASE_VERSION}"
```
Comment (menski) [inline .ci/scripts/release/post-github.sh:5]:
```suggestion
git push origin "clients/go/v${RELEASE_VERSION}"
```

**Review by menski — CHANGES_REQUESTED** [review]
Looks good, I just think you forgot to remove some `echo` debugs from the go tag script

**Comment by npepinpe** [comment]
Oh yeah, I was using that to test as I mentioned :smile:

**Review by menski — APPROVED** [review]
I approved it, but will not trigger a merge yet. Let's discuss how to be sure it works. My proposal would be let's merge it and try to build and 1.3.0-rc1 tomorrow, wdyt?

**Comment by npepinpe** [comment]
:+1: 

It's the last steps of the pipeline, so even if it fails we can still complete it manually. Other than that, we could test it already but it will create a release which we would delete after. We could also do that.

**Comment by menski** [comment]
bors merge

okay let's build RC1 tomorrow

**Comment by Zelldon** [comment]
@menski @npepinpe any reason to not have that on 1.3.x and 1.2x?

**Pull request merged**